tornado/websocket.py

```diff
@@ -265,8 +321,13 @@ def ping(self, data):
         self.ws_connection.write_ping(data)
 
     def on_pong(self, data):
-        """Invoked when the response to a ping frame is received."""
-        pass
+        """Invoked when the response to a ping frame is received.
+
+        If you override this, be sure to call the parent method, otherwise
+        tornado's regular pinging may decide that the connection has dropped
+        and close the websocket.
+        """
+        self.last_pong = IOLoop.current().time()
```

Comment:
It's better to do this outside the overridable method so there's no question about calling the superclass method.